src/errors.rs

```diff
@@ -0,0 +1,56 @@
+use crate::AppInstruction;
+use backtrace::Backtrace;
+use std::panic::PanicInfo;
+use std::sync::mpsc::SyncSender;
+use std::{process, thread};
+
+pub fn handle_panic(info: &PanicInfo<'_>, send_app_instructions: &SyncSender<AppInstruction>) {
+    let backtrace = Backtrace::new();
+    let thread = thread::current();
+    let thread = thread.name().unwrap_or("unnamed");
+
+    let msg = match info.payload().downcast_ref::<&'static str>() {
+        Some(s) => Some(*s),
+        None => info.payload().downcast_ref::<String>().map(|s| &**s),
+    };
```

Comment:
Why not make this `None` if it's `None`?

Reply:
Should we not check for `String` at all? I'm not aware of the specific cases where it returns `&str` and `String`. I just want it to be as thorough as we can, so we don't miss any important info. If `&str` is more common and `String` is comparatively rare, I can refactor it to simply return `None` here.

Reply:
Actually, you're right. :) let's keep it this way